src/driver.rs

```diff
@@ -188,12 +198,30 @@ pub fn main() {
 
     exit(rustc_driver::catch_with_exit_code(move || {
         let mut orig_args: Vec<String> = env::args().collect();
-        let has_sysroot_arg = arg_value(&orig_args, "--sysroot", |_| true).is_some();
+
+        let has_sysroot_arg = |args: &mut Vec<String>| {
+            if arg_value(&args, "--sysroot", |_| true).is_some() {
+                return true;
+            }
+            // https://doc.rust-lang.org/rustc/command-line-arguments.html#path-load-command-line-flags-from-a-path
+            // Beside checking for existence of `--sysroot` from the args on the command line, we need to
+            // check for the arg files that are prefixed with @ as well to be consistent with rustc
+            for arg in args.iter() {
+                if let Some(arg_file_path) = arg.strip_prefix("@") {
+                    if let Ok(lines) = read_lines(arg_file_path) {
```

Comment:
Can't you just use `if let Ok(arg_file) = read_to_string(arg_file_path)` here? And then below `arg_file.lines().flatten()...`?

Also, why is there a `flatten` call on the lines?

Reply:
I was following this example for using `read_lines` https://doc.rust-lang.org/rust-by-example/std_misc/file/read_lines.html#a-more-efficient-approach. I chose this over using `read_to_string` to avoid having to split the string by `\n` because the file can be multiline.

I changed the implementation to use `read_to_string` which allowed me to reuse the `arg_value` function.